addons/account_analytic_analysis/account_analytic_analysis.py

```diff
@@ -93,486 +68,497 @@ def product_id_change(self, cr, uid, ids, product, uom_id, qty=0, name='', partn
+        help="Sum of timesheet lines invoiced for this contract.")
+    remaining_ca = fields.Float(compute='_remaining_ca_calc', string='Remaining Revenue',
+        help="Computed using the formula: Max Invoice Price - Invoiced Amount.",
+        digits=0)
+    revenue_per_hour = fields.Float(compute='_revenue_per_hour_calc', string='Revenue per Time (real)',
+        help="Computed using the formula: Invoiced Amount / Total Time",
+        digits=0)
+    real_margin = fields.Float(compute='_real_margin_calc', string='Real Margin',
+        help="Computed using the formula: Invoiced Amount - Total Costs.",
+        digits=0)
+    theorical_margin = fields.Float(compute='_theorical_margin_calc', string='Theoretical Margin',
+        help="Computed using the formula: Theoretical Revenue - Total Costs",
+        digits=0)
+    real_margin_rate = fields.Float(compute='_real_margin_rate_calc', string='Real Margin Rate (%)',
+        help="Computes using the formula: (Real Margin / Total Costs) * 100.",
+        digits=0)
+    fix_price_invoices = fields.Boolean('Fixed Price')
+    month_ids = fields.Many2one('account_analytic_analysis.summary.month', compute='_analysis_all', string='Month')
+    user_ids = fields.Many2one('account_analytic_analysis.summary.user', compute='_analysis_all', string='User')
+    hours_qtt_est = fields.Float('Estimation of Hours to Invoice')
+    est_total = fields.Float(compute='_sum_of_fields', string="Total Estimation")
+    invoiced_total = fields.Float(compute='_sum_of_fields', string="Total Invoiced")
+    remaining_total = fields.Float(compute='_sum_of_fields', string="Total Remaining", help="Expectation of remaining income for this contract. Computed as the sum of remaining subtotals which, in turn, are computed as the maximum between '(Estimation - Invoiced)' and 'To Invoice' amounts")
+    toinvoice_total = fields.Float(compute='_sum_of_fields', string="Total to Invoice", help=" Sum of everything that could be invoiced for this contract.")
+    recurring_invoice_line_ids = fields.One2many('account.analytic.invoice.line', 'analytic_account_id', string='Invoice Lines', copy=True)
+    recurring_invoices = fields.Boolean(string='Generate recurring invoices automatically')
+    recurring_rule_type = fields.Selection([
+        ('daily', 'Day(s)'),
+        ('weekly', 'Week(s)'),
+        ('monthly', 'Month(s)'),
+        ('yearly', 'Year(s)'),
+        ], 'Recurrency', help="Invoice automatically repeat at specified interval", default='monthly')
+    recurring_interval = fields.Integer('Repeat Every', help="Repeat every (Days/Week/Month/Year)", default=1)
+    recurring_next_date = fields.Date('Date of Next Invoice', default=fields.Date.context_today)
+
+    @api.multi
+    def open_sale_order_lines(self):
+        sale_ids = self.env['sale.order'].search([('project_id','=',self._context.get('search_default_project_id')),('partner_id','in',self._context.get('search_default_partner_id'))])
```

Comment:
Rename 'sale_ids' to 'SaleOrders' because in new API search method returns recordset(s) instead of id(s) !!!